### PR TITLE
Add more features to Group View

### PR DESCRIPTION
**Pull request by dbkr**
List the rooms in a group in their various categories

Includes https://github.com/matrix-org/matrix-react-sdk/pull/1167
Requires https://github.com/vector-im/riot-web/pull/4530

### CODE REVIEW

**Comment by lukebarnard1** [comment]
@dbkr lmk when you want this looked at

**Comment by dbkr** [comment]
@lukebarnard1 this should be good to go now

File: src/components/structures/GroupView.js [hunk 1]
```diff
@@ -18,10 +18,115 @@ import React from 'react';
...
+        {roleHeader}
+        {userNodes}
+    </div>;
+}
+
+const FeaturedUser = React.createClass({
+    displayName: 'FeaturedUser',
+
+    props: {
+        summaryInfo: PropTypes.object.isRequired,
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: It looks like you only ever use the `user_id`, although I'm guessing you'll want more stuff later?

File: src/components/structures/GroupView.js [hunk 2]
```diff
@@ -65,19 +171,108 @@ export default React.createClass({
...
     },
 
+    _onSettingsClick: function() {
+        this.setState({editing: true});
+    },
+
+    _getFeaturedRoomsNode() {
+        const summary = this.state.summary;
+
+        if (summary.rooms_section.rooms.length == 0) return null;
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: See above about null-guarding

File: src/components/structures/GroupView.js [hunk 3]
```diff
@@ -18,10 +18,115 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import MatrixClientPeg from '../../MatrixClientPeg';
 import sdk from '../../index';
+import dis from '../../dispatcher';
 import { sanitizedHtmlNode } from '../../HtmlUtils';
 import { _t } from '../../languageHandler';
+import AccessibleButton from '../views/elements/AccessibleButton';
 
 
+function categoryRoomListNode(rooms, categoryId, category) {
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: This could be a component?

File: src/components/structures/GroupView.js [hunk 4]
```diff
@@ -18,10 +18,115 @@ import React from 'react';
...
+        }
+
+        return <AccessibleButton className="mx_GroupView_featuredThing" onClick={this.onClick}>
+            <RoomAvatar oobData={oobData} width={64} height={64} />
+            <div className="mx_GroupView_featuredThing_name">{roomNameNode}</div>
+        </AccessibleButton>;
+    },
+});
+
+function roleUserListNode(users, roleId, role) {
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: This could be a component

File: src/components/structures/GroupView.js [hunk 5]
```diff
@@ -65,19 +171,108 @@ export default React.createClass({
         });
     },
 
+    _onSettingsClick: function() {
+        this.setState({editing: true});
+    },
+
+    _getFeaturedRoomsNode() {
```
Comment (lukebarnard1) [inline src/components/structures/GroupView.js]: This and the other `_getFeaturedUsersNode` could be a React component.

File: src/components/views/avatars/RoomAvatar.js [hunk 6]
```diff
@@ -126,9 +126,16 @@ module.exports = React.createClass({
     },
 
     getFallbackAvatar: function(props) {
-        if (!this.props.room) return null;
```
Comment (lukebarnard1) [inline src/components/views/avatars/RoomAvatar.js]: Interesting that this was using `this.props` here, when the props are being passed in explicitly. Other functions in `RoomAvatar (`getRoomAvatar` and `getOneToOneAvatar`) also do this and I wonder if they should also be changed.

**Comment by dbkr** [comment]
OK, have added PropTypes.shape, made the functions React components and changed the other this.props -> props, so this should be everything I think

**Comment by lukebarnard1** [comment]
Looks good apart from the lack of run-time null-guarding. PropTypes will only throw warnings all over the place, but the client will still be unusable when the server returns weird stuff. I guess that's acceptable?

**Pull request merged**